smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/tagging/TaggableTrait.java

```diff
@@ -0,0 +1,173 @@
+ */
+public final class TaggableTrait extends AbstractTrait implements ToSmithyBuilder<TaggableTrait> {
+    public static final ShapeId ID = ShapeId.from("aws.api#taggable");
+    private final String property;
+    private final ShapeId tagApi;
+    private final ShapeId untagApi;
+    private final ShapeId listTagsApi;
+    private final Boolean supportsSystemTags;
+
+    private TaggableTrait(Builder builder) {
+        super(ID, builder.getSourceLocation());
+        property = builder.property;
+        tagApi = builder.tagApi;
+        untagApi = builder.untagApi;
+        listTagsApi = builder.listTagsApi;
+        supportsSystemTags = builder.supportsSystemTags;
+    }
+
+    /**
+     * Gets a boolean indicating whether or not the service supports the resource carrying system tags.
+     *
+     * @return Returns true if the service supports the resource carrying system tags.
+     *
+     */
+    public Optional<Boolean> getSupportsSystemTags() {
+        return Optional.ofNullable(supportsSystemTags);
+    }
+
+    /**
+     * Gets the operation referenced that satisfies the ListTagsForResource behavior.
+     *
+     * @return Return the ShapeId of the operation specified for the ListTags API.
+     */
+    public Optional<ShapeId> getListTagsApi() {
+        return Optional.ofNullable(listTagsApi);
+    }
+
+    /**
+     * Gets the operation referenced that satisfies the UntagResource behavior.
+     *
+     * @return Return the ShapeId of the operation specified for the Untag API.
+     */
+    public Optional<ShapeId> getUntagApi() {
+        return Optional.ofNullable(untagApi);
+    }
+
+    /**
+     * Gets the operation referenced that satisfies the TagResource behavior.
+     *
+     * @return Return the ShapeId of the operation specified for the Tag API.
+     */
+    public Optional<ShapeId> getTagApi() {
+        return Optional.ofNullable(tagApi);
+    }
+
+    /**
+     * Gets the rame of the property that represents the tags on the resource, if any.
+     *
+     * @return Return the name of the property that represents the tags of the resource.
+     */
+    public Optional<String> getProperty() {
+        return Optional.ofNullable(property);
+    }
+
+    @Override
+    protected Node createNode() {
+        return new ObjectNode(MapUtils.of(), getSourceLocation())
+                .withOptionalMember("property", getProperty().map(Node::from))
+                .withOptionalMember("tagApi", getTagApi().map(id -> Node.from(id.toString())))
+                .withOptionalMember("untagApi", getUntagApi().map(id -> Node.from(id.toString())))
+                .withOptionalMember("listTagsApi", getListTagsApi().map(id -> Node.from(id.toString())))
+                .withOptionalMember("supportsSystemTags", getSupportsSystemTags().map(Node::from));
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public SmithyBuilder<TaggableTrait> toBuilder() {
```

Comment:
Just return `Builder` directly.

Reply:
This triggers a failure in TaggableTraitTest
```
java.lang.AssertionError: 
Expected: <Trait `aws.api#taggable`, defined at N/A [0, 0]>
     but: was <Trait `aws.api#taggable`, defined at N/A [0, 0]>
```
On a line that looks like this:
```
assertThat(taggableTrait.toBuilder().build(), equalTo(taggableTrait));
```
I don't understand how removing the copying over of the members properly create an equivalent builder, or if a new builder isn't supposed to have the initial properties of the object it came from.

Reply:
Adjusted the return type.